scripts/policy/frameworks/management/agent/main.zeek

```diff
@@ -142,6 +153,52 @@ function send_set_configuration_response(req: Management::Request::Request)
 		g_config_reqid_pending = "";
 	}
 
+event Management::Agent::Runtime::trigger_log_archival(run_archival: bool)
+	{
+	# This is currently final, but could be considered dynamically in the
+	# future if we make this an option.
+	if ( Management::Agent::archive_logs == F )
+		return;
+
+	local ival = Management::Agent::archive_interval;
+
+	# Fall back to the default rotation interval when not set explicitly:
+	if ( ival == 0 secs )
+		ival = Log::default_rotation_interval;
+
+	# Without a default rotation interval individual log streams might still
+	# have rotation enabled, and we could scan all filters to determine
+	# their rotation configuration. But it's not clear that this is
+	# intuitive or needed, since it's uncommon to want rotation for only
+	# some logs. So we simply don't proceed if it's not configured.
+	if ( ival == 0 secs )
+		return;
+
+	local cmd = Management::Agent::archive_cmd;
+
+	if ( cmd == "" )
+		{
+		cmd = join_string_vec(vector(Installation::root_dir, "bin"), "/");
+		cmd = build_path_compressed(cmd, "zeek-archiver");
+		}
+
+	# The logging framework creates the rotation directory on demand, so
+	# only trigger archival when it exists. Don't warn when it does not:
+	# this will often be expected, since in larger clusters many instances
+	# may not run loggers.
+	if ( run_archival && file_size(Log::default_rotation_dir) > 0 )
+		{
+		cmd = fmt("%s -1 %s %s",
+		    cmd, Log::default_rotation_dir,
+		    Management::Agent::archive_dir);
+
+		Management::Log::info(fmt("triggering log archival via '%s'", cmd));
+		system(cmd);
```

Comment:
I'm going back and forth on using `system()` here, but I think it's ok: it'll background the job and if Zeek terminates while it's still in progress, the archive should just retry next time. I'd still say it'd be a nicer to have either the supervisor control the process, or the archiver daemonize itself; but both would be quite a bit of work, so seems ok for now (until we maybe teach the supervisor to run external processes later anyways).